app/models/merchant.rb

```diff
@@ -8,5 +9,109 @@ def self.find_all(params)
   def self.find_one(params)
     find_all(params).first
   end
+
+  def self.most_revenue(params)
+    Merchant.select("merchants.*, SUM(invoice_items.quantity * invoice_items.unit_price) AS revenue")
+    .joins(:invoices)
+    .joins("INNER JOIN invoice_items ON invoices.id = invoice_items.invoice_id")
+    .joins("INNER JOIN transactions ON invoices.id = transactions.invoice_id")
+    .where(transactions: {result: "success"})
+    .group(:id)
+    .order("SUM(invoice_items.quantity * invoice_items.unit_price) DESC")
+    .limit(params[:quantity])
+  end
+
+    
+  def self.most_items(params)
+    Merchant.select("merchants.*, SUM(invoice_items.quantity) as items_sold")
+    .joins(:invoices)
+    .joins("INNER JOIN invoice_items ON invoices.id = invoice_items.invoice_id")
+    .joins("INNER JOIN transactions ON invoices.id = transactions.invoice_id")
+    .where(transactions: {result: "success"})
+    .group(:id)
+    .order("SUM(invoice_items.quantity) DESC")
+    .limit(params[:quantity])
+  end
+
+  def self.most_revenue_between_dates(params)
+    start_date = params[:start]
+    end_date = params[:end]
+    InvoiceItem.select("SUM(invoice_items.quantity * invoice_items.unit_price) AS revenue")
```

Comment:
Because they start with InvoiceItem should they be moved to the InvoiceItem controller?